source/common/upstream/subset_lb.cc

```diff
@@ -421,9 +421,9 @@ SubsetLoadBalancer::PrioritySubsetImpl::PrioritySubsetImpl(const PrioritySet& or
   }
 }
 
-HostSetPtr SubsetLoadBalancer::PrioritySubsetImpl::createHostSet(uint32_t priority) {
+HostSetImpl* SubsetLoadBalancer::PrioritySubsetImpl::createHostSet(uint32_t priority) {
```

Comment:
What's the reasoning behind changing this to return a raw pointer vs. unique_ptr with move semantics?

Reply:
It allows calling host_subset->update() in the subset load balancer without down casts, because the subclass can override the return type.  Not doable if you encapsulate it in the unique pointer.  
If you prefer the casts and explicit ownership I'll switch back.

Reply:
Sorry which is the code that breaks with unique_ptr? In general definitely prefer explicit ownership but curious what is breaking.

Reply:
Or on the other hand it was for some reason which is since factored out since it works now.  Reinstated :-)